libp2p/protocols/kademlia/kademlia.nim

```diff
@@ -68,6 +74,46 @@ proc waitRepliesOrTimeouts(
 
   return (receivedReplies, failedPeers)
 
+proc dispatchPutVal(
+    kad: KadDHT, peer: PeerId, key: Key, val: seq[byte]
+): Future[void] {.
+    async: (raises: [CancelledError, DialFailedError, ValueError, LPStreamError])
+.} =
+  info "dialing", me = kad.rtable.selfId, peer = peer, key = key, val = val[0 .. 3]
```

Comment:
Debug log or trace.
Also, might run into an index out of bounds with `val[0..3]` depending on the length of val

Reply:
I've made a TODO to format the data in a log-friendly form. As this PR moves towards non-draft, the various TODOs will be address/shifted to follow-up PRs, etc.